src/server/cmd/pachd/main.go

```diff
@@ -181,6 +192,28 @@ func doEnterpriseMode(config interface{}) (retErr error) {
 			loggingInterceptor.StreamServerInterceptor,
 		),
 	)
+}
+
+func doReadinessCheck(config interface{}) error {
+	env := serviceenv.InitPachOnlyEnv(serviceenv.NewConfiguration(config))
+	return env.GetPachClient(context.Background()).Health()
+}
+
+func doEnterpriseMode(config interface{}) (retErr error) {
+	defer func() {
+		if retErr != nil {
+			log.WithError(retErr).Print("failed to start server")
```

Comment:
seems like errors from `<-errChan` will end up here, too, not necessarily from starting. goroutine dump is presumably m ore interesting in the unexpected-error-while-running case